libvast/src/type.cpp

```diff
@@ -252,6 +252,12 @@ caf::optional<std::string> record_type::resolve(const offset& o) const {
   return result;
 }
 
+const record_field* record_type::find(std::string_view field_name) const {
+  auto pred = [&](auto field) { return field.name == field_name; };
+  auto i = std::find_if(fields.begin(), fields.end(), pred);
+  return i == fields.end() ? nullptr : &*i;
```

Comment:
We should probably add a warning that the returned pointer may be invalidated when adding new fields to this `record_type`. (it may even be worth thinking about whether we want to disable adding new fields in general, and use a new `mutable_record_type` while building up the record)

Reply:
I've added a note to the interface.

I agree that immutable types are desirable. Once the type system is flatbufferized, I think we will have that useful immutable property.